pkg/cluster/config/v1alpha1/doc.go

```diff
@@ -14,20 +14,9 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package config
-
-// APIVersion is the kubernetes-style API apiVersion for this Config package
-const APIVersion = "kind.sigs.k8s.io/v1alpha1"
-
-// ConfigKind is the kubernetes-style API kind identifier for Config
-const ConfigKind = "Config"
-
-// Kind returns the `kind:` for Config
-func (c *Config) Kind() string {
-	return ConfigKind
-}
-
-// APIVersion returns the `apiVersion:` for Config
-func (c *Config) APIVersion() string {
-	return SchemeGroupVersion.String()
-}
+// Package v1alpha1 implements the v1alpha1 apiVersion of the `kind` Config
+//
+// +k8s:deepcopy-gen=package
+// +k8s:conversion-gen=sigs.k8s.io/kind/pkg/cluster/config/
```

Comment:
(not necessarily for this PR): perhaps we should move this to `sigs.k8s.io/kind/pkg/apis/kind[/v1alpha1]` for consistency with other k8s api packages?

Reply:
would `sigs.k8s.io/kind/pkg/apis/kind[/v1alpha1]` imply `conversion-gen-external-types`?

e.g. how it's done for the kubelet:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/apis/config/v1beta1/doc.go#L18-L19

Reply:
@munnerz you are right, this is not necessary now, but this will be used as soon as v1alpha2 will be added, so if you don't mind I will leave it

Reply:
+1, let's do this in a follow up.

I put it under pkg/cluster/config in an attempt to signal the scope of the config object, let's consider that as we improve the config / API structure, I'd like to leave room to grow in new and unexpected ways ... other container runtimes, possibly CLI tools ... :-)